pkg/ansible/operator/operator.go

```diff
@@ -29,78 +29,84 @@ import (
 	"github.com/operator-framework/operator-sdk/pkg/ansible/flags"
 	"github.com/operator-framework/operator-sdk/pkg/ansible/proxy/controllermap"
 	"github.com/operator-framework/operator-sdk/pkg/ansible/runner"
+	"github.com/operator-framework/operator-sdk/pkg/ansible/watches"
 
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
 	"sigs.k8s.io/controller-runtime/pkg/runtime/signals"
 )
 
+var log = logf.Log.WithName("manager")
+
+const (
+	workerEnvFormat = "WORKER_%s_%s"
+)
+
 // Run - A blocking function which starts a controller-runtime manager
 // It starts an Operator by reading in the values in `./watches.yaml`, adds a controller
 // to the manager, and finally running the manager.
 func Run(done chan error, mgr manager.Manager, f *flags.AnsibleOperatorFlags, cMap *controllermap.ControllerMap) {
-	watches, err := runner.NewFromWatches(f.WatchesFile)
+	watchesSlice, err := watches.Load(f.WatchesFile)
```

Comment:
@LiliC was looking into adding support in the ansible operator for kube-state-metrics, which would require access to the GVKs from `watches.yaml` in `pkg/ansible/run.go`. The helm operator does the watch file parsing in [pkg/helm/run.go](https://github.com/operator-framework/operator-sdk/blob/master/pkg/helm/run.go#L89) and passes the necessary pieces into the controller and metrics registration functions.

Is something like that do-able for the ansible-operator case?

Reply:
I don't know enough about the impact of that sort of change so I'll just ping @shawn-hurley on that.

I do think that we could agree that the changes mentioned would be outside the scope of this PR? I would be happy to implement that move...and I would be happy to see this PR merged :sunglasses:

Reply:
yes I am happy if it's done in a separate PR, but it is something that needs to be done for the custom resource metrics to be enabled.